File: backend/app/schemas/platform.py
from pydantic import BaseModel, Field


class HealthResponse(BaseModel):
    status: str
    service: str
    version: str


class Capability(BaseModel):
    key: str
    name: str
    status: str
    description: str


class PlatformSummary(BaseModel):
    product: str
    architecture: str
    capabilities: list[Capability]


class InfrastructureComponent(BaseModel):
    key: str
    name: str
    status: str
    mode: str
    configured: bool
    endpoint: str | None = None
    details: list[str] = Field(default_factory=list)


class InfrastructureStatus(BaseModel):
    environment: str
    components: list[InfrastructureComponent]


class InfrastructureProbeResult(BaseModel):
    key: str
    name: str
    status: str
    reachable: bool | None = None
    latency_ms: int | None = None
    checked_at: str
    details: list[str] = Field(default_factory=list)


class InfrastructureProbeSummary(BaseModel):
    environment: str
    timeout_seconds: float
    results: list[InfrastructureProbeResult]


class AuthEndpointRead(BaseModel):
    key: str
    name: str
    url: str | None = None
    required: bool = True
    configured: bool


class AuthReadiness(BaseModel):
    mode: str
    provider: str
    issuer: str | None = None
    audience: str | None = None
    status: str
    endpoints: list[AuthEndpointRead] = Field(default_factory=list)
    blockers: list[str] = Field(default_factory=list)
    warnings: list[str] = Field(default_factory=list)
    next_actions: list[str] = Field(default_factory=list)


class AuthorizationResourceRead(BaseModel):
    resource_type: str
    relations: list[str] = Field(default_factory=list)
    permissions: list[str] = Field(default_factory=list)
    notes: list[str] = Field(default_factory=list)


class AuthorizationReadiness(BaseModel):
    mode: str
    provider: str
    status: str
    endpoint: str | None = None
    insecure_transport: bool
    schema_hash: str | None = None
    schema_path: str | None = None
    resources: list[AuthorizationResourceRead] = Field(default_factory=list)
    relationship_count: int
    permission_count: int
    blockers: list[str] = Field(default_factory=list)
    warnings: list[str] = Field(default_factory=list)
    next_actions: list[str] = Field(default_factory=list)


class AuthorizationSchemaRead(BaseModel):
    path: str
    sha256: str
    resource_types: list[str] = Field(default_factory=list)
    relation_count: int
    permission_count: int
    content: str

File: frontend/src/types/platform.ts
export type CapabilityStatus = "not-started" | "foundation" | "partial" | "complete";

export type PlatformCapability = {
  key: string;
  name: string;
  status: CapabilityStatus;
  description: string;
};

export type InfrastructureComponent = {
  key: string;
  name: string;
  status: string;
  mode: string;
  configured: boolean;
  endpoint: string | null;
  details: string[];
};

export type InfrastructureStatus = {
  environment: string;
  components: InfrastructureComponent[];
};

export type InfrastructureProbeResult = {
  key: string;
  name: string;
  status: string;
  reachable: boolean | null;
  latency_ms: number | null;
  checked_at: string;
  details: string[];
};

export type InfrastructureProbeSummary = {
  environment: string;
  timeout_seconds: number;
  results: InfrastructureProbeResult[];
};

export type AuthEndpointRead = {
  key: string;
  name: string;
  url: string | null;
  required: boolean;
  configured: boolean;
};

export type AuthReadiness = {
  mode: string;
  provider: string;
  issuer: string | null;
  audience: string | null;
  status: string;
  endpoints: AuthEndpointRead[];
  blockers: string[];
  warnings: string[];
  next_actions: string[];
};

export type SecretReadinessItem = {
  key: string;
  name: string;
  domain: string;
  status: string;
  required: boolean;
  secret_path_configured: boolean;
  inline_configured: boolean;
  details: string[];
};

export type SecretReadiness = {
  environment: string;
  provider: string;
  status: string;
  configured_count: number;
  vault_path_count: number;
  inline_count: number;
  missing_required_count: number;
  local_default_count: number;
  items: SecretReadinessItem[];
  blockers: string[];
  warnings: string[];
  next_actions: string[];
};

export type AuthorizationResourceRead = {
  resource_type: string;
  relations: string[];
  permissions: string[];
  notes: string[];
};

export type AuthorizationReadiness = {
  mode: string;
  provider: string;
  status: string;
  endpoint: string | null;
  insecure_transport: boolean;
  schema_hash: string | null;
  schema_path: string | null;
  resources: AuthorizationResourceRead[];
  relationship_count: number;
  permission_count: number;
  blockers: string[];
  warnings: string[];
  next_actions: string[];
};

export type AuthorizationSchemaRead = {
  path: string;
  sha256: string;
  resource_types: string[];
  relation_count: number;
  permission_count: number;
  content: string;
};
